Source/lua/usertypes/items.cpp

```diff
@@ -0,0 +1,207 @@
+	    "IMISC_ELIXVIT", IMISC_ELIXVIT,
+	    "IMISC_REJUV", IMISC_REJUV,
+	    "IMISC_FULLREJUV", IMISC_FULLREJUV,
+	    "IMISC_SCROLL", IMISC_SCROLL,
+	    "IMISC_SCROLLT", IMISC_SCROLLT,
+	    "IMISC_STAFF", IMISC_STAFF,
+	    "IMISC_BOOK", IMISC_BOOK,
+	    "IMISC_RING", IMISC_RING,
+	    "IMISC_AMULET", IMISC_AMULET,
+	    "IMISC_UNIQUE", IMISC_UNIQUE);
+	lua.new_enum("SpellID",
+	    "Null", SpellID::Null,
+	    "Firebolt", SpellID::Firebolt,
+	    "Healing", SpellID::Healing,
+	    "Lightning", SpellID::Lightning,
+	    "Flash", SpellID::Flash,
+	    "Identify", SpellID::Identify,
+	    "FireWall", SpellID::FireWall,
+	    "TownPortal", SpellID::TownPortal,
+	    "StoneCurse", SpellID::StoneCurse,
+	    "Infravision", SpellID::Infravision,
+	    "Phasing", SpellID::Phasing,
+	    "ManaShield", SpellID::ManaShield,
+	    "Fireball", SpellID::Fireball,
+	    "Guardian", SpellID::Guardian,
+	    "ChainLightning", SpellID::ChainLightning,
+	    "FlameWave", SpellID::FlameWave,
+	    "DoomSerpents", SpellID::DoomSerpents,
+	    "BloodRitual", SpellID::BloodRitual,
+	    "Nova", SpellID::Nova,
+	    "Invisibility", SpellID::Invisibility,
+	    "Inferno", SpellID::Inferno,
+	    "Golem", SpellID::Golem,
+	    "Rage", SpellID::Rage,
+	    "Teleport", SpellID::Teleport,
+	    "Apocalypse", SpellID::Apocalypse,
+	    "Etherealize", SpellID::Etherealize,
+	    "ItemRepair", SpellID::ItemRepair,
+	    "StaffRecharge", SpellID::StaffRecharge,
+	    "TrapDisarm", SpellID::TrapDisarm,
+	    "Elemental", SpellID::Elemental,
+	    "ChargedBolt", SpellID::ChargedBolt,
+	    "HolyBolt", SpellID::HolyBolt,
+	    "Resurrect", SpellID::Resurrect,
+	    "Telekinesis", SpellID::Telekinesis,
+	    "HealOther", SpellID::HealOther,
+	    "BloodStar", SpellID::BloodStar,
+	    "BoneSpirit", SpellID::BoneSpirit,
+	    "Mana", SpellID::Mana,
+	    "Magi", SpellID::Magi,
+	    "Jester", SpellID::Jester,
+	    "LightningWall", SpellID::LightningWall,
+	    "Immolation", SpellID::Immolation,
+	    "Warp", SpellID::Warp,
+	    "Reflect", SpellID::Reflect,
+	    "Berserk", SpellID::Berserk,
+	    "RingOfFire", SpellID::RingOfFire,
+	    "Search", SpellID::Search,
+	    "RuneOfFire", SpellID::RuneOfFire,
+	    "RuneOfLight", SpellID::RuneOfLight,
+	    "RuneOfNova", SpellID::RuneOfNova,
+	    "RuneOfImmolation", SpellID::RuneOfImmolation,
+	    "RuneOfStone", SpellID::RuneOfStone,
+	    "Invalid", SpellID::Invalid);
+
+	lua.new_usertype<Item>("Item",
+	    "_iSeed", &Item::_iSeed,
+	    "_iCreateInfo", &Item::_iCreateInfo,
+	    "_iAnimFlag", &Item::_iAnimFlag,
+	    "position", &Item::position,
+	    "_iDelFlag", &Item::_iDelFlag,
+	    "selectionRegion", sol::property([](const Item &i) { return static_cast<int>(i.selectionRegion); }),
+	    "_iPostDraw", &Item::_iPostDraw,
+	    "_iIdentified", &Item::_iIdentified,
+	    "_iMagical", sol::property([](const Item &i) { return static_cast<int>(i._iMagical); }),
+	    "_iName", &Item::_iName,

```

Comment:
Do we really want these weird `_i` names in Lua?

Reply:
I don't mind removing those.